specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2020-07-01-preview/extensions.json

```diff
@@ -0,0 +1,808 @@
+            "errorInfo": {
+              "type": "object",
+              "description": "Error information from the Agent - e.g. errors during installation.",
+              "readOnly": true,
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        },
+        "identity": {
+          "$ref": "#/definitions/ConfigurationIdentity",
+          "description": "The identity of the configuration."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "ExtensionInstanceForCreate": {
+      "description": "Object to create a new Extension Instance.",
+      "properties": {
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "Properties of an Extension Instance resource",
+          "properties": {
+            "extensionType": {
+              "type": "string",
+              "description": "Type of the Extension, of which this resource is an instance of.  It must be one of the Extension Types registered with Microsoft.KubernetesConfiguration by the Extension publisher."
+            },
+            "autoUpgradeMinorVersion": {
+              "description": "Flag to note if this instance participates in auto upgrade of minor version, or not.",
+              "type": "boolean"
+            },
+            "releaseTrain": {
+              "description": "ReleaseTrain this extension instance participates in for auto-upgrade (e.g. Stable, Preview, etc.) - only if autoUpgradeMinorVersion is 'true'.",
+              "$ref": "#/definitions/ReleaseTrain"
+            },
+            "version": {
+              "description": "Version of the extension for this extension instance, if it is 'pinned' to a specific version. autoUpgradeMinorVersion must be 'false'.",
+              "type": "string"
+            },
+            "scope": {
+              "description": "Scope at which the extension instance is installed.",
+              "$ref": "#/definitions/Scope"
+            },
+            "configurationSettings": {
+              "description": "Configuration settings, as name-value pairs for configuring this instance of the extension.",
+              "type": "object",
+              "$ref": "#/definitions/ConfigurationSettings"
+            },
+            "configurationProtectedSettings": {
+              "description": "Configuration settings that are sensitive, as name-value pairs for configuring this instance of the extension.",
+              "type": "object",
+              "$ref": "#/definitions/ExtensionProtectedSettings"
+            }
+          }
+        },
+        "identity": {
+          "$ref": "#/definitions/ConfigurationIdentity",
+          "description": "The identity of the configuration."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "ExtensionInstanceForList": {
```

Comment:
why do you need a separate one for LIst call? you should use the ExtensionInstance resource model here. The response of GET and LIST should not be different.

Reply:
LIST response does not return all the properties returned by GET response.  I created a Violation Suppression Request on 8/10/2020 (attached the mail to this PR) explaining this difference.